layers/+spacemacs/spacemacs-defaults/packages.el

```diff
@@ -300,8 +300,10 @@
       ;; Minibuffer history
       (setq savehist-file (concat spacemacs-cache-directory "savehist")
             enable-recursive-minibuffers t ; Allow commands in minibuffers
-            history-length 1000
+            history-length t
+            history-delete-duplicates t
             savehist-additional-variables '(mark-ring
+                                            kill-ring
```

Comment:
This might be a bad idea if you're copy-pasting passwords...

Reply:
So, you suggest removing this functionality?

Reply:
Not necessarily. You could have a toggle for this, off by default, with a a setting and a warning documented.

Reply:
Sure. Will update this.

Reply:
I currently don't know how to add this kind of setting as it requires more logic here and there(by seeing implementation of variables in `dotspacemacs/init`). Can I just remove `kill-ring`?

Reply:
I can do it. It will take some time though.

Reply:
There's no rush. I can come back to this PR and give you some pointers in case you are stuck.

Reply:
Should I force push a single commit everytime?

Reply:
Personally I prefer a single commit, but if it's easier you can push intermediary commits and squash them in the end.

Reply:
I am defining the variables for spacemacs-default layer and not in the core. Is that okay?

Reply:
Not sure, please push your changes and I will comment on the line of code once I see it (and have more context).